### ISSUE TITLE
clean_ami_name improperly squashes space char

### ISSUE THREAD

**Issue by tb3088** [mitchellh/packer#2182]
Maybe back in 2013 a space was illegal but it's in very common use these days.
module: packer/builder/amazon/common/template_funcs.go

**Comment by henrysher** [comment]
according to [AWS API: CreateImage](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateImage.html) 
```
Name
A name for the new image.

Constraints: 3-128 alphanumeric characters, parentheses (()), square brackets ([]), spaces ( ), periods (.), slashes (/), dashes (-), single quotes ('), at-signs (@), or underscores(_)

Type: String

Required: Yes
```

**Comment by tb3088** [comment]
oh well, I guess they need to update the docs. I've done dozens of AMIs with spaces in their names.

**Comment by henrysher** [comment]
@tb3088 i mean, the API document shows these characters are **valid** in AMI name, i suggest to supporting all of them in Packer this time.
```
alphanumeric characters (len: 3-128)
parentheses (()), 
square brackets ([]), 
spaces ( ), 
periods (.), 
slashes (/), 
dashes (-), 
single quotes ('), 
at-signs (@), 
underscores(_)
```

**Comment by catsby** [comment]
Fixed in #2207
